src/ComicK/ComicKParser.ts

```diff
@@ -171,6 +151,66 @@ export const parseChapters = (
     }
 }
 
+const filterUploadersByScore = (chapters: ChapterData[]): ChapterData[] => {
+    const chapterMap = new Map<number, { score: number, chapter: ChapterData }>()
+        for (const chapter of chapters) {
+            const chapNum = Number(chapter?.chap)
+            const chapterScore = chapter.up_count - chapter.down_count
+            if (chapterMap.has(chapNum)) {
+                if (chapterScore > chapterMap.get(chapNum)!.score) {
+                    chapterMap.set(chapNum, { score: chapterScore, chapter: chapter })
+                }
+            } else {
+                chapterMap.set(chapNum, { score: chapterScore, chapter: chapter })
+            }
+        }
+        return Array.from(chapterMap.values(), ((mapValue) => mapValue.chapter))
+}
+
+const filterUploadersByList = (
+    chapters: ChapterData[],
+    uploadersWhitelisted: boolean,
+    aggressiveUploadersFilter: boolean,
+    strictNameMatching: boolean,
+    uploaders: string[]
+): ChapterData[] => {
+    return chapters.filter((chapter) => {
+        const groups = []
+        if (chapter?.group_name) {
+            for (const group of chapter.group_name) {
+                groups.push(group)
+            }
+        }
+
+        if (aggressiveUploadersFilter) {
+            if (uploadersWhitelisted) {
+                // We check if that if the chapter does not have any of the uploaders in the list, we don't push it (we only allow chapters that have all the uploaders in the whitelist)
+                if (!groups.every(group => uploaders.some(uploader => (strictNameMatching && (uploader === group) || (!strictNameMatching && group.toLowerCase().includes(uploader.toLowerCase())))))) {
```

Comment:
this logic is changed compared to previously, because of what I _think_ was a bug with `strictNameMatching` off:
Current behaviour is that say I add "asura" to my blacklist, with `strictNameMatching` off, the group "asurascans" will not be excluded (and vice versa, with whitelisting, "asurascans" will not be included) due to the condition being `uploader.includes(group)`.

This switches it to `group.includes(uploader)`, which means that typing "asura" into the uploaders list will match "asurascans" in the group list. Let me know if this is not desired and I will revert.

Reply:
Compiles fine != good code. Reassigning variables like that makes code hard to maintain

Reply:
did you mean to reply to my other comment where I mentioned that it compiled fine? this comment was about a logic change I made that I _think_ was a bug, so I'm having trouble interpreting your reply with regards to this comment :/